lib/block-template-utils.php

```diff
@@ -98,6 +99,34 @@ function gutenberg_generate_block_templates_export_file() {
 	// Convert to a string.
 	$theme_json_encoded = wp_json_encode( $theme_json_raw, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE );
 
+	// Find any uploaded files.
+	$uploads = wp_upload_dir();
+	$pattern = '/"(' . preg_quote( $uploads['baseurl'], '/' ) . '\/[^"]+)"/';
+	preg_match_all( $pattern, $theme_json_encoded, $matches );
+
+	if ( ! empty( $matches ) ) {
+		$replacement        = '"file:./assets/$1"';
+		$replace_pattern    = '/"' . preg_quote( $uploads['baseurl'], '/' ) . '\/.*?\/([^\/"\s]+)"/';
+		$theme_json_encoded = preg_replace( $replace_pattern, $replacement, $theme_json_encoded );
```

Comment:
Personally, I'm not really sure a regex on the string is the right approach. I wonder if we should be more explicit. We know the structure of the JSON file, we know the properties that are susceptible to contain files. I feel it's better to "traverse" the three and make the changes. (an architecture with a traverse + transformers)

Reply:
Thanks for checking this PR out!

We have most of that logic already thanks to `WP_Theme_JSON_Resolver_Gutenberg::get_resolved_theme_uris`, and the [fonts traversing](https://github.com/WordPress/gutenberg/pull/65019/files#diff-995eb1403df37ca3ca196d6198be899a73dbf5e3aba8fec149c0f1ff1d330935R859) has already been tested.

So not a great leap to move away from regex 👍🏻